plugins/net_plugin/net_plugin.cpp

```diff
@@ -3941,6 +3761,73 @@ namespace eosio {
       }
    }
 
+   digest_type gossip_bp_peers_message::bp_peer::digest() const {
+      digest_type::encoder enc;
+      fc::raw::pack(enc, my_impl->chain_id);
+      fc::raw::pack(enc, producer_name);
+      fc::raw::pack(enc, server_address);
+
+      return enc.result();
+   }
+
+   // called from connection strand
+   void connection::handle_message( gossip_bp_peers_message& msg ) {
+      if (!my_impl->bp_gossip_enabled())
+         return;
+
+      if (!my_impl->validate_gossip_bp_peers_message(msg)) {
+         peer_wlog( this, "bad gossip_bp_peers_message, closing");
+         no_retry = go_away_reason::fatal_other;
+         enqueue( go_away_message( fatal_other ) );
+         return;
+      }
+
+      // valid gossip peer connection
+      is_gossip_bp_connection = true;
+
+      assert(!msg.peers.empty()); // checked by validate_gossip_bp_peers_message()
+      if (msg.peers.size() == 1 && msg.peers[0].server_address.empty()) {
+         // initial message case, send back our entire collection
+         send_gossip_bp_peers_message();
+      } else {
+         bool diff = my_impl->update_gossip_bps(msg);
+         if (diff) { // update, let all our peers know about it
+            send_gossip_bp_peers_message_to_bp_peers();
+         }
+      }
+   }
+
+   // called from connection strand
+   void connection::send_gossip_bp_peers_initial_message() {
+      if (protocol_version < proto_gossip_bp_peers || !my_impl->bp_gossip_enabled())
+         return;
+      auto sb = my_impl->get_gossip_bp_initial_send_buffer();
+      enqueue_buffer(msg_type_t::gossip_bp_peers_message, {}, queued_buffer::queue_t::general, sb, no_reason);
+   }
+
+
+   // called from connection strand
+   void connection::send_gossip_bp_peers_message() {
+      assert(my_impl->bp_gossip_enabled());
+      gossip_buffer_factory factory;
+      const send_buffer_type& sb = my_impl->get_gossip_bp_send_buffer(factory);
+      enqueue_buffer(msg_type_t::gossip_bp_peers_message, {}, queued_buffer::queue_t::general, sb, no_reason);
+   }
+
+   // called from connection strand, thread safe
+   void connection::send_gossip_bp_peers_message_to_bp_peers() {
+      assert(my_impl->bp_gossip_enabled());
+      my_impl->connections.for_each_connection([this](const connection_ptr& c) {
+         gossip_buffer_factory factory;
+         if (this != c.get() && c->is_gossip_bp_connection && c->socket_is_open()) {
```

Comment:
Se we don't want to update the `is_configured_bp_connection` peers?

Reply:
No, manually configured bp connections are not shared with others.